### ISSUE TITLE
automatically refresh statistics widget every 10 seconds to get latest db values

### ISSUE THREAD

**Issue by suryacaprice** [crocodic-studio/crudbooster#259]
automatically refresh statistics widget every 10 seconds to get latest db values ... anyone has any idea how to do it i used this <head> <meta http-equiv="refresh" content="30" /> </head> which refresh the entire page .. i only wanted to update the widgets automatically ... please help me with this .

**Comment by topo79** [comment]
@suryacaprice @jdredondo @fherryfherry any idea, I need the same. if that possible inside CB

**Comment by rallisf1** [comment]
I will be making a guide about how to make it dynamic (add refresh option to the component) but here is a nasty way of achieving the same result:

1. Open /vendor/crocodicstudio/crudbooster/src/views/statistic_builder/index.blade.php
At around line 222 you should find this
```$('.connectedSortable').each(function() {```
2. Move the contents of this loop, except the areaname variable declaration, and create a new function with them above the loop. Your function should look like this:
```
function updateArea(areaname, id_cms_statistics){
	$.get("{{CRUDBooster::mainpath('list-component')}}/"+id_cms_statistics+"/"+areaname,function(response) {       		
		if(response.components) {
			
			$.each(response.components,function(i,obj) {
				$('#'+areaname).append("<div id='area-loading-"+obj.componentID+"' class='area-loading'><i class='fa fa-spin fa-spinner'></i></div>");
				$.get("{{CRUDBooster::mainpath('view-component')}}/"+obj.componentID,function(view) {
					console.log('View For CID '+view.componentID);
					$('#area-loading-'+obj.componentID).remove();
					$('#'+areaname).append(view.layout);
					
				})
			})        				
		}       			
	})
}
```

3. Add your function to the existing loop. The loop should look like this:
```
$('.connectedSortable').each(function() {
	var areaname = $(this).attr('id');
	updateArea(areaname, id_cms_statistics);
})
```

4. Lastly just add an interval function for the area you want
```
var area5Refresh = setInterval(function(){
		$('#area5').empty();
		updateArea('area5', id_cms_statistics);
	}, 30000);
```

The major downside of this is that it will update all the components of the area and there is no easy way to fix it (will need to rebuild the statistics controller and view). So if you have many components and wish to update all the areas i suggest you use high intervals and about 2 seconds apart for each area so the server load is more spread out.

Another downside is that we edited vendor code, which means you have to redo the above after every composer update.

Keep up with my other fixes at #1055

**Comment by topo79** [comment]
hey take a look to my comment on #1055 TY for your help
